File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stack<string> st;
        string subStr = "";
        string ans = "";
        for (int i =0; i<s.length(); i++)
        {
            if(s[i]==' ')
            {
                if(!subStr.empty())
                {
                    st.push(subStr);
                    subStr="";
                }
            }
            else
            {
                subStr += s[i];
            }
        }
        if (!subStr.empty()) 
        {
            st.push(subStr);
        }
        while(st.size()!=1)
        {
            ans += st.top()+" ";
            st.pop();
        }
        ans += st.top();
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        st = []
        subStr = ""
        ans = ""
        n = len(s)

        for ch in s:
            if(ch == ' '):
                if(subStr):
                    st.append(subStr)
                    subStr = ""

            else:
                subStr += ch
        
        if(subStr):
            st.append(subStr) 

        while(len(st) != 1):
            ans = (ans+st.pop()) + " "

        ans = ans + st.pop()

        return ans